src/algorithms/common.jl

```diff
@@ -239,3 +239,28 @@ function optimize(model::AbstractModel, optimizer::AbstractOptimizer, x0, args..
     r = optimize(_model, optimizer, _x0, args...; kwargs...)
     return @set r.minimizer = unflatten(r.minimizer)
 end
+
+"""
+ optimize without x0
+"""
+ function optimize(model::AbstractModel, optimizer::AbstractOptimizer, args...; kwargs...)
+    _model, _, unflatten = tovecmodel(model)
+    r = optimize(_model, optimizer, args...; kwargs...)
+    return @set r.minimizer = unflatten(r.minimizer)
+end
+
+"""
+Clamp the box constraint and evaluate objective function at point x
+"""
+function evaluate!(model::AbstractModel, x::AbstractVector)
```

Comment:
I  don't like this function. It does something simple not requiring a function that's not a standard operation and the name is confusing.

Reply:
project and evaluate